### PR TITLE
[NO QA]: Workspace Feed - Expensify Card

### PR DESCRIPTION
**Pull request by narefyev91**
### Details
Workspace Feed - New Feature initial implementation - Expensify Card.
Feature are currently under Beta access - be sure that you have access to all betas or `expensifyCardFeature`

### Fixed Issues
$ https://github.com/Expensify/App/issues/44301, https://github.com/Expensify/App/issues/44303, https://github.com/Expensify/App/issues/44302
PROPOSAL:

### Tests
1. Open any workspace in which you are an adming
2. Go to More Features
3. Verify that use see a new item "Expensify Card"
4. Switch on -> verify that a new row appears in left menu item
5. Verify that the new menu item is highlighted
6. Click in Expensify Card - Verify that a new screen is opened

Test disabled toggle:
1. Switch on a new feature "Expensify Card"
2. Uncomment line 95 in src/pages/workspace/WorkspaceMoreFeaturesPage.tsx
3. Verify that toggle will have lock icon
4. Click on toggle 
5. Verify that toggle is still in active state
6. Verify that modal is shown
7. Verify that Cancel button is closing modal
8. Verify that go to concierge button will open concierge chat

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Open any workspace in which you are an adming
9. Go to More Features
10. Verify that use see a new item "Expensify Card"
11. Switch on -> verify that a new row appears in left menu item
12. Verify that the new menu item is highlighted
13. Click in Expensify Card - Verify that a new screen is opened

Test disabled toggle:
1. Switch on a new feature "Expensify Card"
2. Uncomment line 95 in src/pages/workspace/WorkspaceMoreFeaturesPage.tsx
3. Verify that toggle will have lock icon
4. Click on toggle 
5. Verify that toggle is still in active state
6. Verify that modal is shown
7. Verify that Cancel button is closing modal
8. Verify that go to concierge button will open concierge chat

### QA Steps
1. Open any workspace in which you are an adming
2. Go to More Features
3. Verify that use see a new item "Expensify Card"
4. Switch on -> verify that a new row appears in left menu item
5. Verify that the new menu item is highlighted
6. Click in Expensify Card - Verify that a new screen is opened

Test disabled toggle:
1. Switch on a new feature "Expensify Card"
2. Uncomment line 95 in src/pages/workspace/WorkspaceMoreFeaturesPage.tsx
3. Verify that toggle will have lock icon
4. Click on toggle 
5. Verify that toggle is still in active state
6. Verify that modal is shown
7. Verify that Cancel button is closing modal
8. Verify that go to concierge button will open concierge chat

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/28352309/3f4d74c9-3584-4bc4-a19b-6b27e01e1e11

https://github.com/Expensify/App/assets/28352309/667fede8-28f6-4d38-850c-3d91402dfa6e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/28352309/cd6e6aa0-ed37-44df-93ed-614f31835ad6

https://github.com/Expensify/App/assets/28352309/97813c08-9e92-4975-ae04-28fb37a1ff02

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/28352309/8c72be31-b132-4ec5-b9d6-92242a0cb362

https://github.com/Expensify/App/assets/28352309/1df5eee0-ace5-4a3a-83a0-027982cf07e8

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/28352309/bdd86bf1-ff95-4228-8330-f4b2ce57c946

https://github.com/Expensify/App/assets/28352309/ba2a3004-7e08-4b87-bddc-e99ced44d1fc

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/28352309/f4875fe7-23ba-4837-9afe-791f0d66057a

https://github.com/Expensify/App/assets/28352309/3fe835b2-e09d-4fd2-9afc-5daa98289a57

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/28352309/0ec5a582-6fab-40a7-8598-57d970b7952d

https://github.com/Expensify/App/assets/28352309/dc06e412-6963-48b0-ad02-885783513c23

</details>

### CODE REVIEW

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/CONST.ts
Comment (mountiny) [inline src/CONST.ts:366]: We are using the WORKSPACE_FEEDS beta for this
```suggestion
```

File: src/libs/Permissions.ts
Comment (mountiny) [inline src/libs/Permissions.ts:54]: Use the `canUseWorkspaceFeeds` for this please
```suggestion
```

**Review by mountiny — DISMISSED** [review]
Looking great, thanks! The beta is already there

**Review by allgandalf — COMMENTED** [review]
Code wise almost LGTM, testing out on platforms

File: src/libs/API/parameters/EnablePolicyExpensifyCardsParams.ts
Comment (narefyev91) [reply src/libs/API/parameters/EnablePolicyExpensifyCardsParams.ts:4]: Yeah good point - will add authToken.
Experiencing the same problem with 404 - and that is a reason with toggling off.
We have this line inside useEffect - which should add/remove left menus
Screenshot 2024-07-01 at 12 33 34
When we fire api call (which is not exists) - field is in pending state - that's why nothing happened immediately 
If i just use here direct key from onyx `policyFeatureStates` and not state `featureStates` it will work
Screenshot 2024-07-01 at 12 35 33
Probably when API will be working - everything will be works as well
@allgandalf

**Review by allgandalf — COMMENTED** [review]
Thanks for the changes @narefyev91 ,Design wise this LGTM, waiting on to hear more about [this](https://github.com/Expensify/App/pull/44549#discussion_r1659763036) issue, will approve after that point cleared

File: src/libs/API/parameters/EnablePolicyExpensifyCardsParams.ts
Comment (trjExpensify) [reply src/libs/API/parameters/EnablePolicyExpensifyCardsParams.ts]: @MariaHCD can you weigh in here to keep it moving? Thanks!

Comment (MariaHCD) [reply src/libs/API/parameters/EnablePolicyExpensifyCardsParams.ts]: Ah, sorry, I missed this.

The BE API endpoint is not implemented yet. We can focus on the frontend elements / flows for now and then once the BE endpoints are implemented, we can update the FE here and retest the flow.

**Review by allgandalf — APPROVED** [review]
This looks good to go, if we ignore the one bug i reported according to this [comment](https://github.com/Expensify/App/pull/44549#discussion_r1661183534), lets get this merged.


Also @narefyev91 , can you please mark this as NO QA :) thanks

**Review by allgandalf — APPROVED** [review]
This looks good to go, if we ignore the one bug i reported according to this https://github.com/Expensify/App/pull/44549#discussion_r1661183534, lets get this merged.

Also @narefyev91 , can you please mark this as NO QA :) thanks

File: src/pages/workspace/WorkspaceMoreFeaturesPage.tsx
Comment (MariaHCD) [inline src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:67]: Similar to https://github.com/Expensify/App/pull/44469/files#r1661049859

`name` will be `card.nameValuePairs.cardTitle` and `limit` will be `card.nameValuePairs.unapprovedExpenseLimit`

But we can update that later

**Review by MariaHCD — APPROVED** [review]
LGTM

**Comment by MariaHCD** [comment]
Merging since @mountiny is OOO today. If there's any improvements we can address them in a follow up PR!

File: src/pages/workspace/WorkspaceMoreFeaturesPage.tsx
Comment (aldo-expensify) [inline src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: What does is expected to happen if we call `Policy.enableExpensifyCard` with `policyID = -1`? I guess the server will throw an error... shouldn't we instead be writing the code like:

```
action: (isEnabled: boolean) => {
  if (!policy?.id) {
    return;
  }
  Policy.enableExpensifyCard(policy.id, isEnabled);
}
```
and disable this button if we don't have a `policy.id` with `disabled: !isEmptyObject(cardsList) || policy?.id,`

Comment (MariaHCD) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: Good point, @aldo-expensify! 

@narefyev91 Can we update to not call Policy.enableExpensifyCard with policyID `-1`?

Comment (narefyev91) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: sure

Comment (narefyev91) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: BTW @MariaHCD @aldo-expensify we have the same logic for all menu items
Screenshot 2024-08-05 at 13 49 49
And probably all those API calls will failed?

Comment (narefyev91) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: A little bit confusing if we just re-write only one menu item - and leave everything else as it is. And why such behaviour was not implemented? Probably we do not block action even if policy id is not exists?

Comment (MariaHCD) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: > And probably all those API calls will failed?

Oh, I didn't realize it was a pattern that was used in the other API calls. But yes, those API calls will all fail because the policyID `-1` is invalid.

Perhaps let's update all of those to not make the API calls if the policyID is not present?

Comment (narefyev91) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: Probably good ask - but we do not have such logic anywhere in the app - that if policy id is not exists - we block user interaction. Probably we need some approvals from @mountiny or some one from top architectural stand point

Comment (aldo-expensify) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: > And why such behaviour was not implemented? Probably we do not block action even if policy id is not exists?

My guess is that the `policy.id` is really never missing at this point so maybe no one has seen this causing an error. My suggestion is more on the lines of writing code that makes more sense when someone reads it, and also in case someone manages to click these buttons when the policy has not loaded (if if if that can really happen)

This looks like a bad use of coalesce `?? -1` just to write less code.

> but we do not have such logic anywhere in the app - that if policy id is not exists - we block user interaction

Are you suggesting that we may want to allow the interaction in this case? if you allow the interaction while the policy is missing, it is for sure just going to cause an error that is probably not handled gracefully, why would we want that?

Comment (narefyev91) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: @aldo-expensify i fully agree with you - that if we do not have policy id - we should just not allow interaction
Yeah i think we allowing interaction in 99% cases - no matter if it will be error or success - that's probably why we have in onyx both optimistic, success and error cases pre-defined.
Long story short - i will glad to add those checks for not allowing interaction with toggles - if policy id does not exists.
@trjExpensify @mountiny @MariaHCD @aldo-expensify can you please give a green light here?

Comment (MariaHCD) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: 🟢 from me

From a backend perspective, I think it makes sense not to call the API unnecessarily when we do not have a policyID - those API calls would just fail - and so would be redundant.

Happy to hear more thoughts on the UX, tho. If the policyID is not present and the call fails, the toggle would just flip back to it's previous state which might be more confusing UX than if the toggle was not interactable at all.

Comment (narefyev91) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: @shawnborton as the UX expert - your thoughts?

Comment (shawnborton) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: Can you summarize the situation in which you are asking for my thoughts? I'm not quite following what we're talking about here.

Comment (narefyev91) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: @shawnborton Problem - we allow to click on any toggle in "More sections" if policy is not exists (for some reason)
Screenshot 2024-08-07 at 14 34 00
We have 3 options here:
1) disable toggle - not give user ability to off/on
2) give user ability to click on toggle (off/on) - but if API failed move toggle state back (on/off) - which will be confusing UX
3) don't do anything here

We currently discussing to fully block user interaction if policy is not exists.

Comment (shawnborton) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: How can you access this screen if the policy doesn't exist though?

Comment (narefyev91) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: Good question - probably not possible based on [src/pages/workspace/withPolicyAndFullscreenLoading.tsx](https://github.com/Expensify/App/blob/main/src/pages/workspace/withPolicyAndFullscreenLoading.tsx)

Comment (shawnborton) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: Got it. I'm not quite following what the situation is exactly that you need advice on, but if I can be helpful, please let me know and we can keep discussing.

Comment (aldo-expensify) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: I think this matters more from a code perspective rather than UI, since it may be impossible for this to happen in a normal user case? 🤷

Comment (narefyev91) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: Ok I will make that changes and push a PR to clean up code

Comment (mountiny) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: Wanted to say that this screen has the policyID in the URL so probably not possible to access.

I do like the clarity though from the code perspective so feel free to clean this up but noting this is lower priority task so try to prioritize other release related work and. Focus on this if you do not have other tasks to handle @narefyev91 thanks!

Thanks @aldo-expensify for raising

Comment (narefyev91) [reply src/pages/workspace/WorkspaceMoreFeaturesPage.tsx:135]: @mountiny yeah - i'm happy to help with anything for upcoming releases - but probably for now have some time to pick this one (until something else will come up). Probably waiting when Third-party workspace card feed doc will reviewed and start on that project

**Pull request merged**
